### ISSUE TITLE
multi-output regression - memory overload

### ISSUE THREAD

**Issue by NikicaJEa** [dmlc/xgboost#10041]
Hi! I was playing around with the recent implementation of multi-output regression. My training dataset has about 2000 observations x 500 observations and the task would be to predict 400 target features. After couple of rounds of training I quickly get memory overload (512 GB RAM). Each round of training would take significantly more RAM than previous. 

There is nothing special in the code that I ran:

```
reg = xgb.XGBRegressor(
    tree_method="hist",
    n_estimators=30,
    n_jobs=1,
    max_depth=8, 
    multi_strategy="multi_output_tree",
    subsample=0.3,
)
```

Was multi-output regression meant to be run on such datasets? Thanks for the help!

**Comment by trivialfis** [comment]
We are still working with a small number of targets, otherwise the gradient histogram can eat up all the memory. See https://github.com/dmlc/xgboost/issues/9043 `Algorithmic Optimization` section for potential options in the future. At the moment, this is a limitation for tree-based multi-target regression.

**Comment by trivialfis** [comment]
Closing in favor of https://github.com/dmlc/xgboost/issues/9043 .
